builder/builder.go

```diff
@@ -88,11 +88,23 @@ func (b *Builder) ScheduleNewBuildWithContainerFile(workflowName string, imageTa
 }
 
 func (b *Builder) ScheduleNewKanikoBuildWithContainerFile(workflowName string, imageTag string, workflowDefinition []byte, build api.BuildSpec) (*api.Build, error) {
-	ib := b.getImageBuilderForKaniko(workflowName, imageTag, workflowDefinition, build.Tasks[0].Kaniko)
+	task := findKanikoTask(build.Tasks)
```

Comment:
I think it is better to return a bool flag on findKanikoTask as well, this way you can avoid proceeding with the code if the task does not exists.

Reply:
That makes sense. We could even call the method within the conditional.

Reply:
@spolti I followed you suggestion and added an additional bool return value that I use to skip the code execution if no Kaniko task is found.